[Date Picker] Fixed issue selecting current day

charlwillia6 opened the pull request:
##### Description
Simple changes to the DatePicker in both the DeploytContractField and SimExchange/Trade to allow the option to pick the current day for the expiration date.  When the current date is selected it will push the TimePicker pat of the DatePicker 60 minutes ahead so the user will not get a warning that the time is not valid.  

##### Checklist
- [x] Linter status: 100% pass
- [x] Changes don't break existing behavior
- [x] Tests coverage hasn't decreased

##### Affected core subsystem(s)
Ui/UX - DatePicker

##### Testing
I have tested this in my own development environment using the Rinkeby Test Network and submitted the results on a development server

##### Refers/Fixes
Fixes: https://github.com/MARKETProtocol/dApp/issues/300

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/MARKETProtocol/dApp?pullRequest=322) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/MARKETProtocol/dApp?pullRequest=322) before we can accept your contribution.<br/><hr/>**Charles Williams** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/MARKETProtocol/dApp?pullRequest=322) it.</sub>

eswarasai commented:
@charlwillia6 -- Looks like there are failing tests. Can you please fix them locally and then update the PR accordingly? Also, if possible please do add relevant tests for the changes you've done in order to increase the code coverage. Thanks!

charlwillia6 commented:
@eswarasai - Sorry about that.  I haven't done testing for React before, so this is all a good learning process for me. I have added some testing for the time check when the current day is selected and I have updated the tests that were already in their for checking the expiration date.  The commits were added to my own fork and should have updated the Travis-CI process.

charlwillia6 commented:
@eswarasai I have tried many times for the Contributor License Agreement, but it continues to say I haven't signed it.  I have two emails associated with my Github account, but they are both included in my account settings.  I am not sure what the issue is, but it says that it is signed when I click on "Details".

pelsasser commented:
@eswarasai - feel free to bypass the CLA here if your read to merge otherwise.

eswarasai commented:
@charlwillia6 -- Looks like you've already signed the CLA since it's showing as passed in all the required checks.

There are couple of issues with the current changes --
- On clearing the date in the input field, it throws below error 
```
Cannot read property 'isSame' of null
return (selectedDate.isSame(now, 'date') ? selectedDate.set(now.toObject()).add(60, 'm') : selectedDate);
```
- Unable to select time for current date. It's always `now + 60 min`

Can you please address the above issues and update the PR accordingly? Thanks!

charlwillia6 commented:
@eswarasai ~ Sorry about that.  I am not really sure how I missed that.  I apologize.  Everything should be working now as attended.

eswarasai commented:
@charlwillia6 -- Thanks for the quick turnaround on the feedback. Overall this is looking good to me.

@pelsasser -- There's a minor issue which is not of priority while changing the selected time of current day. By default, if you select current day, the timer adds 60 min from now. But if you select the calendar and you try to change time manually, although it doesn't allow you to select past time, you can change the time to be less than 60 min as well. Let me know if I should go ahead and merge this PR without worrying about such scenario or not. Thanks!

pelsasser commented:
@eswarasai - that is fine, if someone really wanted to they could create a contract expiring sooner than in an hour.

Thanks @eswarasai and @charlwillia6 for getting this completed.
